tests/test_main.py

```diff
@@ -1,186 +1,201 @@
 import uuid
+from unittest import mock
+import os
 import pytest
-from fastapi import HTTPException
+from typing import Generator
+from unittest.mock import patch, MagicMock
 from fastapi.testclient import TestClient
+from src.schemas import ProjectDetails, Project
+import src.models as models
+from src.service import get_session, delete_project_, update_project_details_, get_projects_, get_project
 from src.main import app
-from typing import Generator
-from src.main import (
-    projects,
-    get_project,
-    create_project_,
-    get_projects_,
-    update_project_details_,
-    delete_project_,
-)
-from src.schemas import Project, ProjectDetails
-
-
-@pytest.fixture
-def client() -> Generator[TestClient]:
-    yield TestClient(app)
 
 
 @pytest.fixture(autouse=True)
-def clear_projects() -> Generator[None]:
-    from src.main import projects
-
-    yield
-    projects.clear()
+def mock_settings_env_vars() -> Generator[None]:
+    mock_env = {
+        "POSTGRES_USER": "test_user",
+        "POSTGRES_PASSWORD": "test_password",
+        "POSTGRES_DB": "test_db",
+        "POSTGRES_PORT": "5432",
+        "POSTGRES_SERVER": "localhost",
+    }
+    with mock.patch.dict(os.environ, mock_env):
+        yield
```

Comment:
since you use default values and env vars are loaded after imports this doesn't do much. Mock the actual variables which hold those